javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserRecordDeclarationTest.java

```diff
@@ -0,0 +1,284 @@
+package com.github.javaparser.symbolsolver.javaparsermodel.declarations;
+
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ParseResult;
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.RecordDeclaration;
+import com.github.javaparser.resolution.declarations.ResolvedReferenceTypeDeclaration;
+import com.github.javaparser.symbolsolver.JavaSymbolSolver;
+import com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade;
+import com.github.javaparser.symbolsolver.model.resolution.TypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.CombinedTypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class JavaParserRecordDeclarationTest {
+
+
+    private final String code = "package x.y; record Test() {}";
+    private TypeSolver typeSolver;
+
+
+    @BeforeEach
+    void setup() {
+        // clear internal caches
+        JavaParserFacade.clearInstances();
+        // init type solver...
+        CombinedTypeSolver combinedTypeSolver = new CombinedTypeSolver();
+        combinedTypeSolver.add(new ReflectionTypeSolver());
+        typeSolver = combinedTypeSolver;
+
+    }
+
+    @Test
+    void testIsRecord() {
+        ParserConfiguration configuration = new ParserConfiguration()
```

Comment:
It seems to me that there could be a bit of refactoring to avoid repeating code.